### DISCUSSION TITLE
Is a MessageContractAttribute required on message classes?

### DISCUSSION THREAD

**Discussion by jecook** [CoreWCF/CoreWCF#546] [Q&A]
I have some old services that have request and response objects without any MessageContractAttribute on them. I see in CoreWCF.Description.TypeLoader that the CreateTypedMessageDescription method gets this attribute and then checks the IsWrapped property. In my case this throws a NRE because the attribute is null.
Can't a missing MessageContractAttribute just default to no wrapping?

**Comment by mconnew** [comment]
I'm presuming your bringing over a working contract from WCF on .NET Framework?  

This might be a bug in our back compat code. Basically we check for attributes in the System.ServiceModel namespace by name and use reflection to get the set properties and create the CoreWCF counterparts. If you are using this back compat capability, can you try switching to using the CoreWCF versions of the attributes and see if you have the same problem?  

Either way, can you open an issue with a simplified minimal contract which has this problem. We do have tests for the MessageContractAttribute so you are using a variation that we aren't testing for. You can just paste the contract as inline code in the issue and don't need to provide a full repro app.

**Reply by jecook** [reply]
I appreciate your quick reply. I'm wondering if you've misunderstood my question, since I don't have a MessageContractAttribute on the offending contracts.

However, I am trying to reproduce with a simplified version, but I haven't yet had enough time to spend on it.

**Reply by mconnew** [reply]
I see I did misunderstand a bit, but trying CoreWCF vs System.ServiceModel namespaced attributes of whatever attributes you are using is still a potentially useful diagnostic step.
